### PR TITLE
Add Discord integration

### PR DESCRIPTION
**Pull request by NorthIsUp**
Add discord regex parsing

- I don't have a developer account so I can't compile this project to test locally

### CODE REVIEW

**Comment by jgoldhammer** [comment]
Thanks for the pull request. It is possible to compile without a (paid) developer account. The same for me. Nevertheless I will try to compile. Do you have a sample discord url for us to test with?

**Comment by jgoldhammer** [comment]
Ping @NorthIsUp...

**Comment by 2Grey** [comment]
@jgoldhammer, examples:

https://discord.com/channels/652753932904562709/778142129314463744
https://discord.com/channels/652753932904562709/778371290372374549

**Native**
discord://discord.com/@me
discord://discord.com/channels/652753932904562709/778142129314463744
discord://discord.com/channels/652753932904562709/778371290372374549

**Comment by 2Grey** [comment]
@NorthIsUp, your code doesn't work for test links.

Patterns should be like:
_For Web_
`#"https://(canary\.)?discord\.com/channels/(\d+/\d+|@me)"#`
_For native_
`#"discord://(canary\.)?discord\.com/channels/(\d+/\d+|@me)"#`

File: MeetingBar/Constants.swift [hunk 1]
```diff
@@ -57,6 +57,8 @@ struct LinksRegex {
     let vonageMeetings = try! NSRegularExpression(pattern: #"https://meetings\.vonage\.com/[0-9]{9}"#)
     let meetStream = try! NSRegularExpression(pattern: #"https://stream\.meet\.google\.com/stream/[a-z0-9-]+"#)
     let around = try! NSRegularExpression(pattern: #"https://meet\.around\.co/[^\s]*"#)
+    let discord = try! NSRegularExpression(pattern: #"https://(canary.)?\.discord\.com/([0-9]+|@me)/[0-9]+"#)
```
Comment (2Grey) [inline MeetingBar/Constants.swift:60]:
```suggestion
    let discord = try! NSRegularExpression(pattern: #"https://(canary\.)?discord\.com/channels/(\d+/\d+|@me)"#)
```

File: MeetingBar/Constants.swift [hunk 2]
```diff
@@ -57,6 +57,8 @@ struct LinksRegex {
...
+    let discord_native = try! NSRegularExpression(pattern: #"discord://(canary.)?\.discord\.com/([0-9]+|@me)/[0-9]+"#)
```
Comment (2Grey) [inline MeetingBar/Constants.swift:61]:
```suggestion
    let discord_native = try! NSRegularExpression(pattern: #"discord://(canary\.)?discord\.com/channels/(\d+/\d+|@me)"#)
```

Comment (2Grey) [reply MeetingBar/Constants.swift:61]: Fix native regexp

File: MeetingBar/Constants.swift (hunk 1 above)
Comment (2Grey) [reply MeetingBar/Constants.swift:60]: Fix web regexp

**Review by 2Grey — CHANGES_REQUESTED** [review]
Change regexp for Web and Native Discord links

**Comment by leits** [comment]
@2Grey, I found this regular expression for discord:
`(https?:\/\/)?(www\.)?(discord\.(gg|io|me|li)|discordapp\.com\/invite)\/.+[a-z]`
Maybe we need to extend the regular expression to handle gg/io/me/li domains?
And maybe you know if /invite is different from /channel? 
Does it make sense to handle /invite?  Or event change regex to handle all endpoints under discord domain?

**Comment by 2Grey** [comment]
@leits,
Maybe something more universal, like
`((https?|discord):\/\/)?(www\.)?(canary\.)?discord(app)?\.([a-zA-Z]{2,})(.+)?`

**Review by leits — DISMISSED** [review]

**Comment by 2Grey** [comment]
@leits, missing Discord image;

Add following code to `StatusBarItemControler.swift:346` class:

```swift
case .some(.discord):
            image = NSImage(named: "Discord-Logo-Color")!
            image?.size = NSSize(width: 16.0, height: 16.0)
```

**Comment by leits** [comment]
Thanks, @2Grey. Already added in separate commit: https://github.com/leits/MeetingBar/commit/93bd428f7fc32da489b9a595fa8269c5df34e37c

**Comment by jgoldhammer** [comment]
Hi @leits  - the discord app protocol pattern was not added, only the https link pattern. Any reason for this?

**Comment by leits** [comment]
Hi, @jgoldhammer. @2Grey сombined them into one 👍

**Pull request merged**